vsintegration/src/vs/FsPkgs/FSharp.LanguageService/servicem.fs

```diff
@@ -2016,12 +2021,32 @@ type FSharpPackage() as self =
     let callback = new ServiceCreatorCallback(CreateIfEnabled)
     
     let mutable mgr : IOleComponentManager = null
+
+    let fsharpSpecificProfileSettings =
+        [| "TextEditor.F#.Insert Tabs", box false
+           "TextEditor.F#.Brace Completion", box true
+           "TextEditor.F#.Indent Style", box 1u |]
     
     override self.Initialize() =
         UIThread.CaptureSynchronizationContext()
+        self.EstablishDefaultSettingsIfMissing()
         (self :> IServiceContainer).AddService(typeof<FSharpLanguageService>, callback, true)
         base.Initialize()
 
+    /// In case custom VS profile settings for F# are not applied, explicitly set them here.
+    /// e.g. 'keep tabs' is the text editor default, but F# requires 'insert spaces'.
+    /// We specify our customizations in the General profile for VS, but we have found that in some cases
+    /// those customizations are incorrectly ignored.
+    member private this.EstablishDefaultSettingsIfMissing() =
+        let settingsManager = this.GetService(typeof<SVsSettingsPersistenceManager>) :?> Microsoft.VisualStudio.Settings.ISettingsManager
+        for settingName,defaultValue in fsharpSpecificProfileSettings do
+            // Only take action if the setting has no current custom value
+            // If cloud-synced settings have already been applied or the user has made an explicit change, do nothing
+            match settingsManager.TryGetValue(settingName) with
+            | Microsoft.VisualStudio.Settings.GetValueResult.Missing, _ ->
+                settingsManager.SetValueAsync(settingName, defaultValue, false) |> ignore
```

Comment:
There is not much use in awaiting this.  It is a plain `Task`, there is no return value to inspect. Even if it fails, there is no particular retry or error handling we would implement.  In theory if `settingsManager` falls out of scope and gets GCed it might cancel its child tasks or something, but practically speaking this won't happen for a shared VS service.

On the Roslyn side they take a [similar approach](https://github.com/dotnet/roslyn/blob/334885aed18184734509a9598d5af3ce8a2cba3b/src/VisualStudio/Core/Impl/Options/AbstractSettingsManagerOptionSerializer.cs#L168) of fire and forget.